app/utils.ts

```diff
@@ -84,3 +84,7 @@ export function decryptCaptchaChallenge(encryptedCaptcha: string): string {
 
   return result;
 }
+
+export function getAppUrl() {
```

Comment:
I think a constant would be better than a function for this case

`const BASE_URL = process.env.APP_URL || "http://localhost:3000"`
```suggestion
const BASE_URL = process.env.APP_URL || "http://localhost:3000"
```